### PR TITLE
Fixup typing regressions

### PR DESCRIPTION
**Pull request by bryevdv**
Some recent PRs introduced a large number of typing issues.

cc @marcinz

### CODE REVIEW

**Comment by bryevdv** [comment]
@fduguet-nv please review https://github.com/nv-legate/cunumeric/pull/471/commits/36bc8ecf1689940614d2022d963ecc5c2948b50e which adds type annotations missing in #254

File: cunumeric/random/bitgenerator.py [hunk 1]
```diff
@@ -12,20 +12,29 @@
...
 
 
 class BitGenerator:
     def __init__(
         self,
-        seed=None,
-        generatorType=BitGeneratorType.DEFAULT,
-        forceBuild=False,
-    ):
+        seed: Union[int, None] = None,
```
Comment (bryevdv) [inline cunumeric/random/bitgenerator.py:34]: Should this be `seed: int = 0` instead? the `seed` value feeds into `task.add_scalar_arg(seed, ty.uint64)` is None acceptable to pass there?

**Comment by bryevdv** [comment]
@fduguet-nv see also https://github.com/nv-legate/cunumeric/pull/471/commits/e7b6a79530ddb3135f2d82529e81b887cd1bfe96 which restores typing in all three thunk modules.

File: cunumeric/random/bitgenerator.py (hunk 1 above)
Comment (fduguet-nv) [reply cunumeric/random/bitgenerator.py:34]: That's the reference
https://numpy.org/doc/stable/reference/random/bit_generators/generated/numpy.random.BitGenerator.html

However, in the current implementation, we only support int as a parameter, or None (which in our case is zero but could be changed)

Comment (bryevdv) [reply cunumeric/random/bitgenerator.py:34]: @manopapad after discussion with @fduguet-nv it seems like it's not clear what to do with None values, presently. Should we raise an explicit error similar to other places where we can't handle certain kinds of values?

Comment (bryevdv) [reply cunumeric/random/bitgenerator.py:34]: We could also just make `seed` a required argument, which diverges from numpy slightly, but has the same practical effect (just a different exception raised)

Comment (fduguet-nv) [reply cunumeric/random/bitgenerator.py:34]: If seed is none, could we set seed to `time.perf_counter_ns()` ? I think that's the most meaningful thing to do here

Comment (bryevdv) [reply cunumeric/random/bitgenerator.py:34]: :+1: that is certainly simple to do, I will push the change. Normally I would also add a unit test with a mock to ensure that `time.perf_counter_ns` is actually called in this case, but I am not sure that will be simple with our oddball testing situation

Comment (bryevdv) [reply cunumeric/random/bitgenerator.py:35]: https://github.com/nv-legate/cunumeric/pull/471/commits/eb1c8649d5007c933afd5649d943ea0a9c13dc3e

**Pull request merged**